pkg/protocol/engine/eviction/state.go

```diff
@@ -323,6 +324,23 @@ func (s *State) PopulateFromStorage(latestCommitmentSlot iotago.SlotIndex) {
 	}
 }
 
+func (s *State) Reset() {
+	rootBlocksToClear := make([]iotago.BlockID, 0)
+	s.rootBlocks.ForEach(func(slot iotago.SlotIndex, storage *shrinkingmap.ShrinkingMap[iotago.BlockID, iotago.CommitmentID]) {
+		if slot > s.lastEvictedSlot {
+			storage.ForEach(func(blockID iotago.BlockID, commitmentID iotago.CommitmentID) bool {
+				rootBlocksToClear = append(rootBlocksToClear, blockID)
+
+				return true
+			})
+		}
+	})
+
+	for _, blockID := range rootBlocksToClear {
+		s.RemoveRootBlock(blockID)
```

Comment:
remove on collection?

Reply:
Foreach grabs a read lock.

Reply:
I changed the code so everywhere where we can actually delete inline, we do it.